packages/discovery-provider/src/tasks/update_delist_statuses.py

```diff
@@ -481,7 +481,7 @@ def revert_delist_status_cursors(self, reverted_cursor_timestamp: float):
         )
 
 
-@celery.task(name="update_delist_statuses", bind=True)
+@celery.task(name="update_delist_statuses", bind=True, soft_time_limit=1, time_limit=2)
```

Comment:
caps the task at a 1s time limit, 2s hard limit.  this raises a time limit exceeded exception if hit and the finally clause releases the lock, if held